mobile/native/src/ln_dlc/mod.rs

```diff
@@ -148,25 +144,22 @@ pub async fn refresh_wallet_info() -> Result<()> {
     Ok(())
 }
 
-pub async fn sync_dlc_channels() -> Result<()> {
+pub async fn sync_node(runtime: &runtime::Handle) {
     let node = state::get_node();
 
-    let runtime = state::get_or_create_tokio_runtime()?;
-
-    runtime
-        .spawn_blocking(move || {
-            if let Err(e) = node.inner.dlc_manager.periodic_chain_monitor() {
-                tracing::error!("Failed to run DLC manager periodic chain monitor task: {e:#}");
-            };
+    if let Err(e) = node.inner.sync_on_chain_wallet().await {
+        tracing::error!("On-chain sync failed: {e:#}");
+    }
 
-            if let Err(e) = node.inner.dlc_manager.periodic_check() {
-                tracing::error!("Failed to run DLC manager periodic check: {e:#}");
-            };
-        })
-        .await
-        .expect("task to complete");
+    runtime.spawn_blocking(move || {
+        if let Err(e) = node.inner.dlc_manager.periodic_chain_monitor() {
+            tracing::error!("Failed to run DLC manager periodic chain monitor task: {e:#}");
+        };
 
-    Ok(())
+        if let Err(e) = node.inner.dlc_manager.periodic_check() {
```

Comment:
Same as above. I think the convenience of combining all the syncs is worth the cost of needlessly syncing every now and then.

In any case, we can simply modify our fork of `rust-dlc` to not include the `periodic_check` in the sub-channel one. That was only added out of convenience so that LN-DLC consumers would not need to manage two separate periodic checks.

Reply:
I think we should do that 👍

Reply:
Done here: https://github.com/p2pderivatives/rust-dlc/commit/ec6f1f7. Added to https://github.com/get10101/10101/pull/2238.